.github/workflows/pr-build.yml

```diff
@@ -0,0 +1,98 @@
+name: "PR Build"
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+    name: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - id: setup-java-8
+        name: Setup Java 8
+        uses: actions/setup-java@v2
+        with:
+          distribution: adopt
+          java-version: 8
+      - uses: burrunan/gradle-cache-action@v1
+        name: restore gradle cache
+        with:
+          gradle-version: wrapper
+          job-id: pr-build
+          remote-build-cache-proxy-enabled: false
+      - name: ./gradlew lint
+        run: ./gradlew lint
+      - uses: burrunan/gradle-cache-action@v1
+        name: ./gradlew assemble
+        with:
+          arguments: --stacktrace assemble
+          gradle-version: wrapper
+          job-id: java-contrib
+          remote-build-cache-proxy-enabled: false
+      - name: ./gradlew test
+        uses: burrunan/gradle-cache-action@v1
+        id: gradle-test
+        timeout-minutes: 5
+        with:
+          arguments: --stacktrace --max-workers 2 test
+          gradle-version: wrapper
+          job-id: java-contrib
+          remote-build-cache-proxy-enabled: false
+      - uses: actions/upload-artifact@v2
+        name: Save unit test results
+        if: always()
+        with:
+          name: test-results
+          path: contrib/jmx-metrics/build/reports/tests/test
+      - name: ./gradlew build
+        uses: burrunan/gradle-cache-action@v1
+        id: gradle-build
+        timeout-minutes: 5
+        with:
```

Comment:
build runs assemble and test, we'd generally just have one step running `build`

Reply:
I liked the distinction between assemble and test to reduce the surface area of issues, with a final build verification step.  I've combined the test and build steps, if the remaining distinction is ok with you (can update otherwise).